keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+[ReferenceGrant](https://gateway-api.sigs.k8s.io/api-types/referencegrant/) was
+developed by the [Gateway API subproject](https://gateway-api.sigs.k8s.io/) to
+enable certain object references to cross namespaces. More recently, it has also
+been [used by
+sig-storage](https://kubernetes.io/blog/2023/01/02/cross-namespace-data-sources-alpha/)
+to enable cross-namespace data sources.
+
+This KEP proposes moving ReferenceGrant from its current
+`gateway.networking.k8s.io` API group to a new `grants.authorization.k8s.io` API
+group.
+
+## Motivation
+
+Now that it's clear that ReferenceGrant is useful beyond just Gateway API, it
+would be good to formalize this model in a more neutral home, ideally sig-auth.
+At this point, each project that wants to enable cross-namespace references has
+to choose between introducing a dependency on Gateway API and creating a new
+resource that would largely duplicate ReferenceGrant. Both options would lead to
+confusion for Kubernetes users.
```

Comment:
Small nit: I think it might improve readability to reverse these sentences for future readers. I also suggested some changes here too for your considerations:

```suggestion
Any project that wants to enable cross-namespace references currently has to choose
between introducing a dependency on Gateway API's ReferenceGrant or creating a
new API that would be partially redundant (leading to confusion for users).

Recent interest between SIGs has made it clear that ReferenceGrant is wanted for use
cases other than Gateway API. We would like to move ReferenceGrant to a neutral home
(ideally, under sig-auth) in order to make it the canonical API for managing references
across namespaces.
```

Reply:
Thanks, I agree, updating.